src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs

```diff
@@ -246,36 +247,32 @@ private static string GetIndentedXmlString(XmlDocument doc)
         /// returns a path to the project file which defines the benchmarks
         /// </summary>
         [PublicAPI]
-        protected virtual FileInfo GetProjectFilePath(Type benchmarkTarget, ILogger logger)
+        protected virtual FileInfo GetProjectFilePath(BenchmarkCase benchmark, ILogger logger)
         {
+            var benchmarkTarget = benchmark.Descriptor.Type;
+
             if (!GetSolutionRootDirectory(out var rootDirectory) && !GetProjectRootDirectory(out rootDirectory))
             {
                 logger.WriteLineError(
                     $"Unable to find .sln or .csproj file. Will use current directory {Directory.GetCurrentDirectory()} to search for project file. If you don't use .sln file on purpose it should not be a problem.");
                 rootDirectory = new DirectoryInfo(Directory.GetCurrentDirectory());
             }
```

Comment:
This can be moved to the ProjectLocator also.

Reply:
GetSolutionRootDirectory is also used in DotNetCliGenerator.cs:54 which is the only reason I didn't move them to the locator. I'll see what can be done.

Reply:
I've let some of the code stay in DotNetCliGenerator as I have no project to test with. I'll focus on CsProj for now.